namespaces/init.go

```diff
@@ -82,11 +82,19 @@ func Init(container *libcontainer.Config, uncleanRootfs, consolePath string, pip
 		}
 	}
 
-	if err := setupNetwork(container, networkState); err != nil {
-		return fmt.Errorf("setup networking %s", err)
-	}
-	if err := setupRoute(container); err != nil {
-		return fmt.Errorf("setup route %s", err)
+	cloneFlags := GetNamespaceFlags(container.Namespaces)
+
+	if (cloneFlags & syscall.CLONE_NEWNET) == 0 {
+		if len(container.Networks) != 0 || len(container.Routes) != 0 {
```

Comment:
I think that combining all the if clauses won't be that good if we add more things to what we do in a particular namespace. So, we can have two level ifs like here, but just take out the else. If we have gotten past the the ifs, then we know that it is safe to call the setup for a particular namespace.